src/proxies/groupsapi.js

```diff
@@ -0,0 +1,119 @@
+const platformClient = require('purecloud-platform-client-v2');
+const retry = require('@lifeomic/attempt').retry;
+
+let groupsMap = new Map();
+/*
+    The getGroup() function will make a call to the platformClient.getGroups() call passing in target page number.
+*/
+const getGroup = async (pageNum) => {
+  let opts = {
+    pageSize: 50,
+    pageNumber: pageNum,
+  };
+
+  let apiInstance = new platformClient.GroupsApi();
+
+  try {
+    results = await apiInstance.getGroups(opts);
+    return results;
+  } catch (e) {
+    console.log(
+      `Error while retrieving group for page number: ${pageNum}: ${JSON.stringify(
+        e,
+        null,
+        '\t'
+      )}`
+    );
+    return null;
+  }
+};
+
+/*
+  The getGroups() call is going to retrieve all of the groups within the org and then map the logical name (e.g. the human readable name) to the Genesys Cloud Group GUID.  Later when we have to
+  add a user to a group we are going to look up the Genesys Cloud GUID for the group by is logical name.  Note:  You have to be aware of whether or not the API call might expect to returns the results using
+  pagination and code accordingly to.  In the code below, I paginate over each page, looking up the values for all pages.  I flatten the results into 1 big list and then build a map containing just the logical 
+  name for the group and the GUID.
+*/
+async function getGroups() {
+  let groups = [];
+
+  //Do the first call and push the results to an array
+  group = await getGroup(1);
+  group != null ? groups.push(group.entities) : null;
+
+  //If the count is greater then 1 then go through and look up the result of the pages.
+  if (group != null && group.pageCount > 1) {
```

Comment:
This condition should be nested in the if statement above.
Also `i++` is more common than i += 1;

```suggestion
  for(let i = 2; i <= group.pageCount; i++) {
```